### PR TITLE
sequence: Add new test

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Checks if given value is a list or not.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/sequence.yml
docs/docsite/rst/user_guide/playbooks_tests.rst
lib/ansible/plugins/test/core.py
test/integration/targets/test_core/tasks/main.yml

### CODE REVIEW

File: docs/docsite/rst/user_guide/playbooks_tests.rst [hunk 1]
```diff
@@ -380,6 +380,25 @@ The following tasks are illustrative of the tests meant to check the status of t
...
+    - assert:
+        that:
+          - ['1', '2'] is sequence
+          - not 'string_value' is sequence
+          - 'string_value' is sequence(include_strings=True)
+
+This would result in::
+
+    { "changed": false, "msg": "All assertions passed" }
+
```
Comment (bcoca) [inline docs/docsite/rst/user_guide/playbooks_tests.rst:401]: maybe add a  note explaining include_strings (cause strings are lists by default in Python)

**Comment by flowerysong** [comment]
Jinja [already has](https://jinja.palletsprojects.com/en/latest/templates/#jinja-tests.sequence) a `sequence` test with different behaviour, so this has the potential to break existing playbooks.

**Comment by sivel** [comment]
I agree with @flowerysong, we should not overload this since jinja2 already has this.

An example that already works: `value is sequence and value is not string` to omit strings.

Modifying the tests that this PR adds to use the built in jinja2 functionality:

```
    - name: Assert sequence tests work
      assert:
        that:
          - not ("string" is sequence and "string" is not string)
          - '["true", "false"] is sequence'
          - '"string" is sequence'
```

Closing per above.

**Comment by Akasurde** [comment]
PR with just tests - https://github.com/ansible/ansible/pull/74897

**Pull request closed without merging**